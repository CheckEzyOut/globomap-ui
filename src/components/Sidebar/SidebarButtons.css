.sidebar-buttons {
  padding: 0 10px;
}

.sidebar-buttons button {
  position: relative;
  display: block;
  width: 100%;
  text-align: center;
  background: transparent;
  color: white;
  border: 0;
  cursor: pointer;
  font-size: 1.6em;
  height: 120px;
  border-top: 1px solid rgba(255, 255, 255, 0.1);
  outline: none;
}

.sidebar-buttons button:last-child {
  border-bottom: 1px solid rgba(255, 255, 255, 0.1);
}

.sidebar-buttons button:hover {
  color: #2eb78e;
}

.sidebar-buttons button:hover:before,
.sidebar-buttons button.active:before {
  content: '';
  position: absolute;
  top: 50%;
  left: 0;
  background-color: #2eb78e;
  height: 40px;
  margin-top: -20px;
  margin-left: -7px;
  width: 4px;
}
