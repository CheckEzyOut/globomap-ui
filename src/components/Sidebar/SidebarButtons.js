/*
Copyright 2018 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setTab } from '../../redux/modules/tabs';
import { Tab } from '../';
import {
  clearNodes,
  clearCurrentNode } from '../../redux/modules/nodes';
import './SidebarButtons.css';

export class SidebarButtons extends Component {

  handleSearch = () => {
    const nodeList = this.props.nodeList;
    const currentTab = this.props.currentTab;
    if (nodeList.length > 0 && currentTab === 'search') {
      this.props.clearNodes();
      this.props.clearCurrentNode();
    }
  }

  render() {
    return (
      <div className="sidebar-buttons">
        <Tab tabKey="search">
          <button onClick={() => this.handleSearch()}>
            <i className="fa fa-search"></i>
          </button>
        </Tab>
        <Tab tabKey="map">
          <button>
            <i className="fa fa-sitemap"></i>
          </button>
        </Tab>
        <Tab tabKey="reports">
          <button>
            <i className="fa fa-print"></i>
          </button>
        </Tab>
        <Tab tabKey="automap">
          <button>
            <i className="fas fa-project-diagram"></i>
          </button>
        </Tab>
      </div>
    );
  }

}

function mapStateToProps(state) {
  return {
    nodeList: state.nodes.nodeList,
    currentTab: state.tabs.currentTab
  };
}

export default connect(
  mapStateToProps,
  {
    setTab,
    clearNodes,
    clearCurrentNode
  }
)(SidebarButtons);
