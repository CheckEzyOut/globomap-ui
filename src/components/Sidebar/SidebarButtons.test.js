import React from 'react';
import { shallow } from 'enzyme';
import { SidebarButtons } from './SidebarButtons';

describe("SidebarButtons", function() {

  it('should render SidebarButtons component', () => {
    const component = shallow(<SidebarButtons />);
    expect(component).toMatchSnapshot();
  });

});
