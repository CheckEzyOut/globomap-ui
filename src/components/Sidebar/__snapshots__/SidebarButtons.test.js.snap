// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarButtons should render SidebarButtons component 1`] = `
<div
  className="sidebar-buttons"
>
  <Connect(Tab)
    tabKey="search"
  >
    <button>
      <i
        className="fa fa-search"
      />
    </button>
  </Connect(Tab)>
  <Connect(Tab)
    tabKey="map"
  >
    <button>
      <i
        className="fa fa-sitemap"
      />
    </button>
  </Connect(Tab)>
  <Connect(Tab)
    tabKey="favorites"
  >
    <button>
      <i
        className="fa fa-star"
      />
    </button>
  </Connect(Tab)>
</div>
`;
